src/systemcmds/tests/test_parameters.cpp

```diff
@@ -157,14 +178,186 @@ bool ParameterTest::ResetAllExcludesWildcard()
 	return ret;
 }
 
+bool ParameterTest::exportImport()
+{
+	bool ret = true;
+
+	param_t test_params[] = {p0, p1, p2, p3};
+
+	// set all params to corresponding param_t value
+	for (auto p : test_params) {
+		if (param_type(p) == PARAM_TYPE_INT32) {
+			int set_val = p;
+			param_set_no_notification(p, &set_val);
+
+			ret = ret && _assert_parameter_int_value(p, p);
+		}
+
+		if (param_type(p) == PARAM_TYPE_FLOAT) {
+			float set_val = p;
+			param_set_no_notification(p, &set_val);
+
+			ret = ret && _assert_parameter_float_value(p, (float)p);
+		}
+	}
+
+	// save
+	if (param_save_default() != PX4_OK) {
+		PX4_ERR("param_save_default failed");
+		return false;
+	}
+
+	// zero all params and verify, but don't save
+	param_control_autosave(false);
+
+	for (auto p : test_params) {
+		if (param_type(p) == PARAM_TYPE_INT32) {
+			int set_val = 0;
+			param_set_no_notification(p, &set_val);
+			ret = ret && _assert_parameter_int_value(p, 0);
+		}
+
+		if (param_type(p) == PARAM_TYPE_FLOAT) {
+			float set_val = 0;
+			param_set_no_notification(p, &set_val);
+			ret = ret && _assert_parameter_int_value(p, 0);
+		}
+	}
+
+	// load saved params
+	if (param_load_default() != PX4_OK) {
+		PX4_ERR("param_save_default failed");
+		ret = true;
+	}
+
+	// check every param
+	for (auto p : test_params) {
+		if (param_type(p) == PARAM_TYPE_INT32) {
+			ret = ret && _assert_parameter_int_value(p, p);
+		}
+
+		if (param_type(p) == PARAM_TYPE_FLOAT) {
+			ret = ret && _assert_parameter_float_value(p, (float)p);
+		}
+	}
+
+	return ret;
+}
+
+bool ParameterTest::exportImportAll()
+{
+	// backup current parameters
+	const char *param_file_name = PX4_ROOTFSDIR "/fs/microsd/param_backup";
+	int fd = open(param_file_name, O_WRONLY | O_CREAT, PX4_O_MODE_666);
+
+	if (fd < 0) {
+		PX4_ERR("open '%s' failed (%i)", param_file_name, errno);
+		return false;
+	}
+
+	int result = param_export(fd, false);
+	close(fd);
+
+
+	bool ret = true;
+
+	int N = param_count();
+
+	// set all params to corresponding param_t value
+	for (param_t p = 0; p < N; p++) {
+
+		if (param_type(p) == PARAM_TYPE_INT32) {
+			int set_val = p;
+			param_set_no_notification(p, &set_val);
+
+			ret = ret && _assert_parameter_int_value(p, p);
+		}
+
+		if (param_type(p) == PARAM_TYPE_FLOAT) {
+			float set_val = p;
+			param_set_no_notification(p, &set_val);
+
+			ret = ret && _assert_parameter_float_value(p, (float)p);
+		}
+	}
+
+	// save
+	if (param_save_default() != PX4_OK) {
+		PX4_ERR("param_save_default failed");
+		return false;
+	}
+
+	// zero all params and verify, but don't save
+	param_control_autosave(false);
```

Comment:
Can you call that at the beginning of the test?

Reply:
The idea was to keep it on during the first pass to replicate normal usage.

Reply:
I tend towards disabling it for all tests, because it's an async thing and not explicitly part of the tests, meaning if for example the auto-saving timeout changes, the tests could start to fail, due to unrelated reasons.